_posts/2019-03-08-code-quality-inspection-for-clojure-with-sonarqube.md

```diff
@@ -0,0 +1,194 @@
+---
+layout: post
+title: Code quality inspection for Clojure using SonarQube  
+author: hjhamala
+excerpt: Clojure has good libraries for code quality analyzing and vulnerabilities detection. I introduce the support for Kibit, ancient-clj, lein-nvd and Cloverage in existing Clojure SonarQube plugin. 
+tags:
+- SonarQube
+- Clojure
+- ClojureScript
+- Code quality
+- Vulnerabilities detection
+- Continuous inspection
+---
+Building good and secure code has never been easy. Programmers make mistakes during the development. Programs use external libraries which may have  bugs and vulnerabilities. New vulnerabilities are found which forces to update existing programs.
+
+Software development have many ways to battle against bugs. For example testing, code reviews and pair programming can help us to make better and safer programs. Fortunately we may also use code analyzers which automate the testing and vulnerability finding.
+
+[**Clojure**](https://clojure.org/) is not immune for bugs or vulnerabilities. Clojure as a dynamic language leads to situations where some problems may occur in runtime which in static languages are found out in compile time. The Clojure community have produced many useful tools such as linters and security scanners.
```

Comment:
"immune to"
"Clojure being a dynamic language" or some restructuring
"occur at runtime"
"found out at compile time" or "found during compiling" or something similar
"community has produced"